### PR TITLE
[MRG] add GraphLabels.link_tag_and_label to CPython API

### PR DESCRIPTION
**Pull request by ctb**
This PR:

* adds the C++ `LabelHash.link_tag_and_labels(tag, label)` function into the Python API for `GraphLabels` objects. 
* updates the `add_tag` and `add_stop_tag` methods on CPython class `Hashgraph` descendants to take either k-mer hashes OR strings as inputs, and supplies explicit tests for this behavior.
* fixes a problem where `GraphLabels` objects *didn't actually inherit from `Hashgraph`* despite being labeled as such in the CPython inheritance hierarchy, and it brings the Cython wrapping/definitions into concordance with the CPython hierarchy.
* add/fixes tests relying on the inheritance hierarchy.

Checklist:

- [x] Is it mergeable?
- [x] `make test` Did it pass the tests?
- [x] `make clean diff-cover` If it introduces new functionality in
  `scripts/` is it tested?
- [x] `make format diff_pylint_report cppcheck doc pydocstyle` Is it well
  formatted?
- [x] Did it change the command-line interface? Only backwards-compatible
  additions are allowed without a major version increment. Changing file
  formats also requires a major version number increment.
- [x] For substantial changes or changes to the command-line interface, is it
  documented in `CHANGELOG.md`? See [keepachangelog](http://keepachangelog.com/)
  for more details.
- [x] Was a spellchecker run on the source code and documentation after
  changes were made?
- [x] Do the changes respect streaming IO? (Are they
  tested for streaming IO?)

### CODE REVIEW

**Comment by ctb** [comment]
I am having an interesting problem where 'GraphLabel' objects don't appear to inherit 'Hashgraph' methods. Specifically, if I change `lb.graph.add_tag(...)` to `lb.add_tag(...)` the tests hang. Any thoughts @camillescott @luizirber @betatim?

(See [full source context here](https://github.com/dib-lab/khmer/pull/1758/files#diff-c1e7c162a38040d60d12f758a33faceeR208))

**Comment by ctb** [comment]
I think this is ready for review @betatim @camillescott @standage @luizirber. Tempting fate, I will say that this is a fairly simple and straightforward set of fixes ;)

File: src/khmer/_cpy_graphlabels.cc [hunk 1]
```diff
@@ -391,6 +397,30 @@ labelhash_get_tag_labels(khmer_KGraphLabels_Object * me, PyObject * args)
...
+    }
+    if (!ht_convert_PyObject_to_HashIntoType(tag_o, tag,
+            labelhash->graph)) {
+        return NULL;
+    }
+
+    labelhash->link_tag_and_label(tag, label);
+
+    Py_INCREF(Py_None);
+    return Py_None;
```
Comment (standage) [inline src/khmer/_cpy_graphlabels.cc]: What is the purpose of these two lines? Reading between the lines, it looks like its simply making the default behavior explicit. @ctb

File: tests/test_labelhash.py [hunk 2]
```diff
@@ -201,21 +201,58 @@ def test_get_tag_labels():
...
+    lb = GraphLabels(20, 1, 1)
+
+    tag = 173473779682
+    kmer = lb.reverse_hash(tag)
+    lb.add_tag(kmer)
+    lb.link_tag_and_label(kmer, 1)
+
+    labels = lb.get_tag_labels(tag)       # <-- use 'tag' instead of 'kmer'
+    assert len(labels) == 1
+    assert labels.pop() == 1
```
Comment (standage) [inline tests/test_labelhash.py]: Tests are simple enough that even *I* can understand what's going on! :-)

**Review by standage — APPROVED** [review]
Looks good. Should hopefully be simple to merge with @camillescott's open PR on cythonizing hashgraph stuffs.

**Pull request merged**